front/lib/auth.ts

```diff
@@ -934,11 +934,16 @@ export class Authenticator {
 
     let groups: GroupResource[] = [];
     if (authType.groupIds.length > 0 && workspace) {
-      // Create a temporary authenticator for fetching groups
+      // Temporary authenticator used solely to fetch the group resources. We
+      // grant it the `admin` role so that it can read any group in the
+      // workspace, irrespective of membership. The returned authenticator
+      // (see below) will still use the original `authType.role`, so this
+      // escalation is confined to the internal bootstrap step and does not
+      // leak outside of this scope.
       const tempAuth = new Authenticator({
         workspace,
         user,
-        role: authType.role,
+        role: "admin",
```

Comment:
Noob question: so the error we had was that groupsResult was not ok and raising with [this error](https://app.datadoghq.eu/logs?query=%22%5BAuthenticator.fromJSON%5D%20Failed%20to%20fetch%20groups%22&agg_m=count&agg_m_source=base&agg_t=count&cols=host%2Cservice&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1753196498425&to_ts=1753200098425&live=true). 


Should we instead of returning an err in `GroupResource.fetchByIds` just exclude the groups we can't read? 
Why do we need to fake an admin auth to fake listing the groups of the user? I'm sure what you're doing is right but I don't really understand 🙏🏻